app/assets/javascripts/discourse-common/addon/utils/dom-utils.js

```diff
@@ -0,0 +1,51 @@
+function isWindow(obj) {
+  return obj != null && obj === window;
+}
+
+function scrollTop(element) {
+  return element
+    ? element.scrollTop
+    : window.pageYOffset || document.documentElement.scrollTop;
+}
+
+function offset(element) {
```

Comment:
The vast majority of our usage here is to get the top offset. Maybe this should simply return the `topOffset`. I think in this PR, we're only using that. If eventually we need to use the `left` offset as well, we can add another helper for it. 

Your call, though, i don't feel strongly about it.

Reply:
Thanks but I will keep the full object for now, calling `.top` is not much IMO